tests/tests_deployment/test_jupyterhub_api.py

```diff
@@ -52,6 +56,38 @@ def test_check_default_roles_added_in_keycloak():
     role_names = [role["name"] for role in client_roles]
     assert "allow-app-sharing-role" in role_names
     assert "allow-read-access-to-services-role" in role_names
+    assert "allow-group-directory-creation-role" in role_names
+
+
+@pytest.mark.filterwarnings(
+    "ignore:.*auto_refresh_token is deprecated:DeprecationWarning"
+)
+@pytest.mark.filterwarnings("ignore::urllib3.exceptions.InsecureRequestWarning")
+def test_check_directory_creation_scope_attributes():
+    client_role = get_keycloak_client_role(
+        client_name="jupyterhub", role_name="allow-group-directory-creation-role"
+    )
+    assert client_role["attributes"]["component"][0] == "shared-directory"
+    assert client_role["attributes"]["scopes"][0] == "write:shared-mount"
+
+
+@pytest.mark.filterwarnings(
+    "ignore:.*auto_refresh_token is deprecated:DeprecationWarning"
+)
+@pytest.mark.filterwarnings("ignore::urllib3.exceptions.InsecureRequestWarning")
+def test_groups_with_mount_permissions():
+    client_role = get_keycloak_client_role(
+        client_name="jupyterhub", role_name="allow-group-directory-creation-role"
+    )
+    client_details = get_keycloak_client_details_by_name(client_name="jupyterhub")
+    role_groups = get_keycloak_role_groups(
+        client_id=client_details["id"], role_name=client_role["name"]
+    )
+    assert list([group["path"] for group in role_groups]) == [
+        "/developer",
+        "/admin",
+        "/analyst",
+    ]
```

Comment:
Do a set comparison instead of list as the ordered might not be guaranteed, that's why the test is failing.

Reply:
the funny thing was that I remember addressing this already but It seems I never committed it